multicluster/cmd/multicluster-controller/memberclusterannounce_webhook.go

```diff
@@ -47,13 +46,6 @@ func (v *memberClusterAnnounceValidator) Handle(ctx context.Context, req admissi
 		return admission.Errored(http.StatusBadRequest, e)
 	}
 
-	ui := req.UserInfo
-	_, saName, err := serviceaccount.SplitUsername(ui.Username)
-	if err != nil {
-		klog.ErrorS(err, "Error getting ServiceAccount name", "request", req)
-		return admission.Errored(http.StatusBadRequest, err)
```

Comment:
Okay. I think it's better to keep it, too. But when a new MemberClusterAnnounce is added to the leader cluster.  We cannot get the ServiceAccount. Maybe we can add a new field in the struct ?

Reply:
But when I tried to check the ServiceAccount in the webhook, I found that we could not get the ServiceAccount from the Secert. For example: 
```yaml
apiVersion: v1
data:
  ca.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUMvakNDQWVhZ0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFWTVJNd0VRWURWUVFERXdwcmRXSmwKY201bGRHVnpNQjRYRFRJeU1EWXlPVEF5TXpBMU5sb1hEVE15TURZeU5qQXlNekExTmxvd0ZURVRNQkVHQTFVRQpBeE1LYTNWaVpYSnVaWFJsY3pDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTTVaCjI0SkVIcjVVRjNNRFYrdHFiekd5c0hLNDVVWCtNUE91V1NrbmdTU0FQUWpwbFVXS0pVKzl5VWJFa0ZFUFpFOWIKUUhISlp3N1k0cHdraTM0Y3JGcmNnemVaUnlqRUdaNVJ6WGFXVWZzTk9ZVFQ0WUY4L1pxdTRObXVub2NJcTFGOAppWnZ2WTEvcWRkYThHS1MzenZUVkwyRGhsT1BDL3gyT3JLcmJqREdTcGNXcXVVRU16dGJGdXVHSnVxRXh2U3hECkFnNjN2NzZxQzNvYWYxV2Y1RHlya3gyNzM1N3NBdEp2VEZVRVkyVHN6UmVvVlVOY01PdFAyUXVzZFZGNWU3dlQKdjZMNzFmb0ErTjhvdmM3Rk9IUlV0amlKeHk0OGZsUmt5Y1VIeG1qeWFndW9WL0NpUnd0c2xya0FJcnJQQ1B2RQpFSHVBUjF2ekdKeEFOU1E1VjljQ0F3RUFBYU5aTUZjd0RnWURWUjBQQVFIL0JBUURBZ0trTUE4R0ExVWRFd0VCCi93UUZNQU1CQWY4d0hRWURWUjBPQkJZRUZPdVY5NEQwbVRMZGQxVzhnRVh6NlhJUk9MdTdNQlVHQTFVZEVRUU8KTUF5Q0NtdDFZbVZ5Ym1WMFpYTXdEUVlKS29aSWh2Y05BUUVMQlFBRGdnRUJBQVV1WmI0UEh1MkNIWlNaN3N6Kwp3a2NrS1k4NmtwbVJDa1Z2bE1BM3BZaHp0ZnpKZ1BISFhqVzRxVkZxK1BwOXhEaUlXUU80d1ZwVlp6amg5NmxPCk1JWXo0UjZiMzBMZTQ3UnhBNnlFUCs2TXMxR1l1NjA3eTdjWXErd1cyRHRkaEhlQnUyMlpoYVF4dXhiUllHUXMKWXpZSnZwcEdZWVZpUlNQL2FsY1pVSlFtbjRRQjYrcWpscTFzbHdyUHVnWnl4ekdQMDVIUXFXUmFGdVpZM1QvUgp3SEZHYmkwUUYydDhUZGRqckJJTzFZMWJjMVZ0QnN2VjRXWW9MWTFGL3dMOUkrZG1NYXhaT2ttNnRvZWhzNTd6Cm5kWjBsaDRvY3lya2Z6SU81OFFkWEU0YVMraVhZNlZXa2lyTllpOWNRUUVuR2ZMbVVIbGwvZmVBaU9HWktrRXMKNDJRPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
  namespace: YW50cmVhLW11bHRpY2x1c3Rlcg==
  token: ZXlKaGJHY2lPaUpTVXpJMU5pSXNJbXRwWkNJNklrdE9iMXBJY0dkcGFGRTNMV05uZGtaM1VYbFBjVEZsTkVGb1VqSkRVREJYYzNWNU9FTTRTRFJDWW5NaWZRLmV5SnBjM01pT2lKcmRXSmxjbTVsZEdWekwzTmxjblpwWTJWaFkyTnZkVzUwSWl3aWEzVmlaWEp1WlhSbGN5NXBieTl6WlhKMmFXTmxZV05qYjNWdWRDOXVZVzFsYzNCaFkyVWlPaUpoYm5SeVpXRXRiWFZzZEdsamJIVnpkR1Z5SWl3aWEzVmlaWEp1WlhSbGN5NXBieTl6WlhKMmFXTmxZV05qYjNWdWRDOXpaV055WlhRdWJtRnRaU0k2SW0xbGJXSmxjaTFoWTJObGMzTXRkRzlyWlc0aUxDSnJkV0psY201bGRHVnpMbWx2TDNObGNuWnBZMlZoWTJOdmRXNTBMM05sY25acFkyVXRZV05qYjNWdWRDNXVZVzFsSWpvaWJXVnRZbVZ5TFdGalkyVnpjeTF6WVNJc0ltdDFZbVZ5Ym1WMFpYTXVhVzh2YzJWeWRtbGpaV0ZqWTI5MWJuUXZjMlZ5ZG1salpTMWhZMk52ZFc1MExuVnBaQ0k2SW1NNU9UYzNaakEwTFRNd04yUXROR0kyTWkxaVpEQmxMV00wTURBM1pqVTRNR1UyWmlJc0luTjFZaUk2SW5ONWMzUmxiVHB6WlhKMmFXTmxZV05qYjNWdWREcGhiblJ5WldFdGJYVnNkR2xqYkhWemRHVnlPbTFsYldKbGNpMWhZMk5sYzNNdGMyRWlmUS5Bamk3bmZQUDkyeFhrenlLUVh1MXNWSmFUODNyQ3lFazZROENGNHZrMzRaNjFXVGtLdVI0QzgtMldjNERLQ2hreXFvWm9taEsyajFPekRoc2VKY0xPMEstVkdSZ2xOcXRpcUxVV3BlX01PR1RXM2MtR2xjSktDOVpoQmtjX09idTJTSGdTNkl4QlQyblZDUW1CR2tnemo0RG1LUHhWSTBkT3FJVDdpdUdFX2Z3bzdXNHlobjJwX3AtMmFuUkpPdS1EOGVnT1lUMmp5T1hYeHI5TUlSWVlZOGNLWkxyc3BYeHF6RGFUcG9Rb1JDMjI4U25LRUZLcXdnbHV5bWtGdkdWOGJ5LVA1Q0JaNTFDRnozVGdNVHlKbnlJcmtuem1DaDI3QkRlcThITFJFRVZjcGtVd3F6emtFLWduYXZzUHZ4OHcxUmh1bmtpNnloVldaRm9SVnVrZ2c=
kind: Secret
metadata:
  name: member-access-token
  namespace: kube-system
type: Opaque
```

Maybe an annotation in the Secret and add a new field in the MemberClusterAnnounce, so that we can check the ServiceAccount in the webhook.

Reply:
Fixed. Keep the SA check and add a ServiceAccount unit test.